### PR TITLE
Add logging utilities

### PR DESCRIPTION
**Pull request by ignaloidas**

### CODE REVIEW

File: b_lambda_layer_common/source/python/util/logging.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import logging
+from typing import Optional
+
+
+class LoggingManager:
+    '''
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:6]: This is not how docstrings are used in python. Please do some research about python docstrings.

File: b_lambda_layer_common/source/python/util/logging.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    def __init__(self,
+                 *,
+                 log_level: int = logging.INFO,
+                 formatter: Optional[logging.Formatter] = None):
+        self.log_level = log_level
+        self.formatter = formatter
+
+    def setup_logging(self):
+        '''
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:20]: Always use double quotes for docstrings.

File: b_lambda_layer_common/source/python/util/logging.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    '''
+    Class responsible for common logging setup
+
+    :param log_level: Default log level set for logging
+    :param formatter: Default formatter set for logging
+    '''
+    def __init__(self,
+                 *,
+                 log_level: int = logging.INFO,
+                 formatter: Optional[logging.Formatter] = None):
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:15]: Bad formatting. Add add new line before ")" symbol.

File: b_lambda_layer_common/source/python/util/logging.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class LoggingManager:
+    '''
+    Class responsible for common logging setup
+
+    :param log_level: Default log level set for logging
+    :param formatter: Default formatter set for logging
+    '''
+    def __init__(self,
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:12]: Bad formatting. Add add new line after "(" symbol.

File: b_lambda_layer_common/source/python/util/logging.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            root_logger = logging.getLogger()
+            root_logger.setLevel(self.log_level)
+            for handler in root_logger.handlers:
+                handler.setLevel(self.log_level)
+                if self.formatter:
+                    handler.setFormatter(self.formatter)
+            root_logger.info("Set up logging")
+        except:
+            # We don't want this to ever fail. If it fails though, it should fail as loud as it can without disruption.
+            logging.exception("Failed to set up logging")  # Prints out exception info
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:33]: Use comments above the line, not in the same line.

File: b_lambda_layer_common/source/python/util/logging.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        self.formatter = formatter
+
+    def setup_logging(self):
+        '''
+        Set up the logging with the default parameters.
+        '''
+        try:
+            root_logger = logging.getLogger()
+            root_logger.setLevel(self.log_level)
+            for handler in root_logger.handlers:
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:26]: For better readability, please add more new lines to separate "for" loop.

File: b_lambda_layer_common/source/python/util/logging.py (hunk 1 above)
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:6]: Add docstring not just for a constructor but for class too.

**Review by laimonassutkus — CHANGES_REQUESTED** [review]
Pull request can not be approved as there are some changes that have to be implemented.

**Comment by ignaloidas** [comment]
Implemented the requested changes

**Review by laimonassutkus — CHANGES_REQUESTED** [review]
Since you are an employee of Biomapas and a member of an internal IT team you should be able to access this guideline: https://biomapas.atlassian.net/wiki/spaces/IT/pages/143589439/Docstrings Please follow this guideline regarding structure of docstrings.

File: b_lambda_layer_common/source/python/util/logging.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import logging
+from typing import Optional
+
+
+class LoggingManager:
+    """
+    Class responsible for common logging setup
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:7]: Sentences should end with a dot.

File: b_lambda_layer_common/source/python/util/logging.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    """
+
+    def __init__(
+            self,
+            *,
+            log_level: int = logging.INFO,
+            formatter: Optional[logging.Formatter] = None
+        ) -> None:
+        """
+        :param log_level: Default log level set for logging
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/util/logging.py:17]: Sentences should end with a dot.

**Review by laimonassutkus — CHANGES_REQUESTED** [review]
Sentences should end with a dot.

**Review by laimonassutkus — APPROVED** [review]

**Review by laimonassutkus — CHANGES_REQUESTED** [review]
I think there should be changes in HISTORY.md file too.

**Review by laimonassutkus — APPROVED** [review]

**Pull request merged**